src/components/IconInput/IconInput.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from 'react-emotion';
+import { size } from 'polished';
+import Input from '../Input';
+
+import { themePropType } from '../../util/shared-prop-types';
+
+const LEFT = 'left';
+const RIGHT = 'right';
+
+const iconBaseStyles = ({ theme }) => css`
+  label: icon-input__icon;
+  position: absolute;
+  ${size(theme.spacings.mega)};
+  top: 50%;
+  transform: translateY(-50%);
+`;
+
+const iconLeftStyles = ({ theme, iconPosition }) =>
+  iconPosition === LEFT &&
+  css`
+    label: icon-input__icon--left;
+    left: ${theme.spacings.kilo};
+    right: auto;
+  `;
+
+const iconRightStyles = ({ theme, iconPosition }) =>
+  iconPosition === RIGHT &&
+  css`
+    label: icon-input__icon--right;
+    right: ${theme.spacings.kilo};
+    left: auto;
+  `;
+
+const inputLeftStyles = ({ theme, iconPosition }) =>
+  iconPosition === LEFT &&
+  css`
+    label: icon-input__input--left;
+    padding-left: calc(
+      ${theme.spacings.kilo} + ${theme.spacings.mega} + ${theme.spacings.kilo}
+    );
+  `;
+
+const inputRightStyles = ({ theme, iconPosition }) =>
+  iconPosition === RIGHT &&
+  css`
+    label: icon-input__input--right;
+    padding-right: calc(
+      ${theme.spacings.kilo} + ${theme.spacings.mega} + ${theme.spacings.kilo}
+    );
+  `;
+
+/**
+ * Renders inputs or selects that have an icon overlay. Takes the icon as a
+ * render prop.
+ */
+const IconInput = ({ children, iconPosition, theme, ...props }) => {
+  const iconClassName = cx(
+    iconBaseStyles({ theme }),
+    iconLeftStyles({ theme, iconPosition }),
+    iconRightStyles({ theme, iconPosition })
+  );
+
+  const inputClassName = cx(
+    inputLeftStyles({ theme, iconPosition }),
+    inputRightStyles({ theme, iconPosition })
+  );
+
+  return (
+    <Input {...props} className={inputClassName}>
+      {children({ className: iconClassName })}
```

Comment:
Didn't this previously forward a disabled class, too? So that the icon could be made transparent, when the input is disabled?

Reply:
This is not necessary anymore, because the disabled styles are applied to the wrapper `div` and cascade down to the icon.

Reply:
Aha! 🤦🏼‍♂️